src/tblite/xtb/calculator.f90

```
@@ -204,17 +207,19 @@ subroutine add_basis(calc, mol, param, irc)
 end subroutine add_basis
 
 
-subroutine add_ncoord(calc, mol, param)
+subroutine add_ncoord(calc, mol, param, error)
    !> Instance of the xTB evaluator
    type(xtb_calculator), intent(inout) :: calc
    !> Molecular structure data
    type(structure_type), intent(in) :: mol
    !> Parametrization records
    type(param_record), intent(in) :: param
+   !> Error handling
+   type(error_type), allocatable, intent(out) :: error
 
-   if (allocated(param%hamiltonian%cn)) then
-      call new_ncoord(calc%ncoord, mol, cn_type=param%hamiltonian%cn)
-   end if
+   integer :: cn_count_type
+
+   call new_ncoord(calc%ncoord, mol, cn_count_type=param%hamiltonian%cn, error=error)
```

Does removing the `if (allocated(param%hamiltonian%cn))` check yield the same behavior as before? I think, before, it just silently skipped creating the CN if no parameters are given. Does it now just fail? Do we want that?